main/src/ca/uwaterloo/flix/language/errors/TypeError.scala

```diff
@@ -48,17 +48,75 @@ object TypeError {
          |
          |${code(loc, "unable to generalize the type scheme.")}
          |
+         |The declared type does not match the inferred type:
+         |
          |  Declared: ${cyan(FormatScheme.formatSchemeWithoutConstraints(declared))}
          |  Inferred: ${magenta(FormatScheme.formatSchemeWithoutConstraints(inferred))}
          |""".stripMargin
     }
 
-    /**
-      * Returns a formatted string with helpful suggestions.
-      */
+    def explain(formatter: Formatter): Option[String] = Some(
+      """The declared type is more polymorphic than the inferred type.
+        |
+        |Typically:
+        |
+        |  (a) the declared type signature is incorrect, or
+        |  (b) the expression body is incorrect.
```

Comment:
Using the word "incorrect" does not respect neutral or friendly messages. Maybe we should use "does not match" or something else.